src/runtime/pipeline/pipeline_struct.h

```diff
@@ -0,0 +1,170 @@
+  std::unordered_map<int, OutputBindings> output_binding_map;
+  OutputMap& operator=(const OutputMap& output) {
+    output_binding_map = output.output_binding_map;
+    return *this;
+  }
+
+  /*!\brief This function is used to verify whether OutputMap is loaded successfully.
+   * \return Return true to indicate that this class has not been successfully loaded.
+   */
+  bool Empty() { return output_binding_map.empty(); }
+  /*! \brief Global output is the final outputs of pipeline, this function is used to
+   *   get how many global outputs are in this Outputmap
+   *  \return Number of global outputs.
+   */
+  size_t GetGlobalOutputNum(void) const {
+    size_t num_output = 0;
+    for (auto bindings : output_binding_map) {
+      num_output += bindings.second.IsGlobalOutput() ? 1 : 0;
+    }
+    return num_output;
+  }
+
+  /*!
+   * \brief Create output binding map from JSONReader.
+   * \param reader Json reader.
+   */
+  void Load(dmlc::JSONReader* reader) {
+    reader->BeginArray();
+    while (reader->NextArrayItem()) {
+      std::string key;
+      reader->BeginObject();
+      int output_idx = -1;
+      OutputBindings binding;
+      while (reader->NextObjectItem(&key)) {
+        if (key == "output_idx") {
+          reader->Read(&output_idx);
+        }
+        if (key == "dependent") {
+          reader->Read(&binding);
+        }
+      }
+      ICHECK(output_idx >= 0);
+      output_binding_map[output_idx] = binding;
+    }
+  }
+};
+/*!
+ * \brief Binding or dependency information of each module output interface.
+ */
+struct PipelineConfig {
+  /*!\brief The module index is the key, this variable records all module pipeline configuration
+   * information.
+   */
+  std::unordered_map<int, OutputMap> config;
+  OutputMap& operator[](int key) {
+    ICHECK(config.find(key) != config.end());
+    return config[key];
+  }
+
+  void Insert(int key, const OutputMap& map) { config[key] = map; }
+  /*!
+   * \brief Get the number of global outputs that is the outputs of entire pipeline.
+   * \return How much output does the entire pipeline have.
+   */
+  size_t GetGlobalOutputNum() const {
+    size_t num_output = 0;
+    for (auto mod_output : config) {
+      num_output += mod_output.second.GetGlobalOutputNum();
+    }
+    return num_output;
+  }
+};
+/*!
+ * \brief The informations used to initialize the graph executor module, the information
+ *  come from the export library function call.
+ */
+struct ModuleInformation {
+  ModuleInformation(const std::string& lib, const std::string& json, const std::string& params,
+                    const std::string& device)
+      : lib_name(lib), json_name(json), params_name(params), dev(device) {}
+  ModuleInformation() { ; }
+  std::string lib_name;
+  std::string json_name;
+  std::string params_name;
+  std::string dev;
+};
+/*! The Module information of each module.The 'int' is module index. */
+using ModuleConfig = std::unordered_map<int, ModuleInformation>;
```

Comment:
`ModuleInformation` and `ModuleConfig` have two similar names but they are different things.. To avoid confusion, please come up with a better name for `ModuleInformation`. The name should clearly tell what "information" this struct contains.

Reply:
changed into GraphModuleLoadInfo